src/index.d.ts

```diff
@@ -0,0 +1,256 @@
+// Type definitions for bonescript 0.6
+// Project: https://github.com/jadonk/bonescript
+// Definitions by: Troy W. <https://github.com/troywweber7>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as SerialPort from 'serialport';
+import * as VError from 'verror';
+
+// OBJECTS
+export const bone: BoneObj;
+
+// FUNCTIONS - TODO obliterate all "any" types where possible! --- TWW
+// https://github.com/jadonk/bonescript#system
+
+export function getPlatform(callback?: (error: ErrorType, platform: any) => void): any;
+
+export function getEeproms(callback?: (error: ErrorType, eeproms: any) => void): any;
+
+export function echo(data: any, callback?: (error: ErrorType, data: any) => void): any;
+
+export function readTextFile(filename: string, callback?: (error: ErrorType, data: any) => void): any;
+
+export function writeTextFile(filename: string, data: any, callback?: (error: ErrorType, data: any) => void): void;
+
+export function writeCModule(filename: string, data: any, callback?: (error: ErrorType, data: any) => void): void;
+
+export function setDate(date: any, callback?: (error: ErrorType, data: any) => void): void;
```

Comment:
I don't think this should be returning any 'data', just an error status. The 'date' should be a string type with specific formatting.